src/virtual_dom/vlist.rs

```diff
@@ -104,22 +105,67 @@ impl VDiff for VList {
 
         // Process children
         let mut lefts = self.children.iter_mut();
-        let mut rights = rights.drain(..);
-        loop {
-            match (lefts.next(), rights.next()) {
-                (Some(left), Some(right)) => {
-                    previous_sibling = left.apply(parent, previous_sibling.as_ref(), Some(right));
+        let has_keys = if let Some(first) = rights.first() {
```

Comment:
I would like to see a solution that handles keys well but also falls back gracefully to the previous ordered approach. 

---

How about calling a new function that re-orders `rights` optimally, and then follow the same code we had before?

`reorder_rights(...)`:
* Takes `lefts` and `rights` as inputs
* Compares keys between the two and reorders `rights` as needed
* Note: it cannot reorder `lefts`

Then, it will be easier to test and optimize in the future 😄

Reply:
I think you could get down to one `loop` if you only remove the `rights` that have keys and then whenever you come across a `left` that doesn't have a matching keyed `right`, use the next node in the remaining `rights`

Reply:
Switched to using a vector for rights without keys.